atc/creds/vault/manager.go

```diff
@@ -84,6 +87,59 @@ func (manager *VaultManager) MarshalJSON() ([]byte, error) {
 	})
 }
 
+func to_string(s interface{}) string {
+	if s == nil {
+		return ""
+	}
+	return s.(string)
+}
+
+func to_bool(s interface{}) bool {
+	if s == nil {
+		return false
+	}
+	return s.(bool)
+}
+
+func to_duration(s interface{}, defaultValue time.Duration) time.Duration {
+	if s == nil {
+		return defaultValue
+	}
+	return s.(time.Duration)
+}
+
+func (manager *VaultManager) UnmarshalJSON(text []byte) error {
```

Comment:
I considered this. The reason why I decided to write `UnmarshalJSON` is that I want to Vault config to be consistent with CLI options. For example, in CLI, there is a option `--vault-client-token`, if I use json tag, then in yaml it should be:

    config: 
       auth:
          client_token: xxx

I want one level config:

    config:
       client_token: xxx

so that, user just needs to check `concourse web --help`, then he knows how to config in yaml.

Reply:
I was hoping there was a `json:",inline"` of sorts but it doesn't seem to exist, so :+1: for now. I think this kind of goes hand-in-hand with my other comment so we can skip over this too.